include/psmove.h

```diff
@@ -1027,43 +1153,209 @@ ADDCALL psmove_get_orientation(PSMove *move,
 ADDAPI void
 ADDCALL psmove_reset_orientation(PSMove *move);
 
+/**
+ * \brief Set the orientation fusion algorithm to use.
+ *
+ * There are currently the following orientation filter algorithms available:
+ * OrientationFusion_MadgwickIMU - Classic MadgwickIMU: Gyro integration + Gravity correction
+ *  - 1st Least expensive
+ *  - Con: Suffers from pretty bad drift
+ * OrientationFusion_MadgwickMARG - Classic MadgwickMARG: Gyro integration + Gravity/Magnetometer correction
+ *  - 2nd least expensive
+ *  - Con: Suffers from slow drift about the yaw
+ * OrientationFusion_ComplementaryMARG - Gyro integration blended with optimized Gravity/Magnetometer alignment
+ *  - Suffers no drift
+ *  - Con: Most expensive algorithms of the three (but not horrendously so)
+
+ * \param move A valid \ref PSMove handle
+ * \param fusion_type The orientation fusion algorithm denoted by the \ref PSMoveOrientation_Fusion_Type enum
+ **/
+ADDAPI void
+ADDCALL psmove_set_orientation_fusion_type(PSMove *move, enum PSMoveOrientation_Fusion_Type fusion_type);
 
 /**
- * \brief Reset the magnetometer calibration state.
+ * \brief Set the transform used on the calibration data in the psmove_get_transform_<sensor>_... methods
  *
- * This will reset the magnetometer calibration data, so they can be
- * re-adjusted dynamically. Used by the calibration utility.
+ * This method sets the transform used to modify the calibration vectors returned by:
+ * - psmove_orientation_get_magnetometer_calibration_direction()
+ * - psmove_orientation_get_gravity_calibration_direction()
  *
- * \ref move A valid \ref PSMove handle
+ * The transformed calibration data is used by the orientation filter to compute 
+ * a quaternion (see \ref psmove_orientation_get_quaternion) that represents 
+ * the controllers current rotation from the "identity pose".
+ * 
+ * Historically, the "identity pose" bas been with the controller laying flat
+ * with the controller pointing at the screen. However, now that we have a
+ * calibration step that record the magnetic field direction relative to 
+ * gravity it makes more sense to make the identity pose with the controller 
+ * sitting vertically since it's more stable to record that way. 
+ *  
+ * In order to maintain reverse compatibility, this transform defaults to rotating
+ * the vertically recorded calibration vectors 90 degrees about X axis as if the 
+ * controller was laying flat during calibration.
+ *
+ * Therefore, if you want a different "identity pose" then the default,
+ * use this method to set a custom transform.
+ *
+ * There are the following transforms available:
+ * - k_psmove_identity_pose_upright - "identity pose" is the controller standing upright
```

Comment:
Or if the transforms are just used here, maybe make a function that takes an `enum` instead, and have an additional method where the user can pass in a custom matrix.